### PR TITLE
Added support for Forge's new brewing API. Closes #221

### LINKED ISSUE #221: [Request] Support for Forge's new brewing system
It's been added recently and is included in the latest recommended version of Forge.
Here's the PR that got merged: https://github.com/MinecraftForge/MinecraftForge/commit/0373920f994562e850a3aad474b0e45585436f4d

It still maintains vanilla functionality so the current brewing recipe viewing you have should stay in place.

### PR DESCRIPTION
**Pull request by ganymedes01**
The old system is still in place, all I did was add in the new stuff where it needed.

### CODE REVIEW

File: src/codechicken/nei/recipe/BrewingRecipeHandler.java [hunk 1]
```diff
@@ -83,30 +97,50 @@ public String getRecipeName() {
...
-            if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.precursorPotion.item, ingredient))
-                arecipes.add(new CachedBrewingRecipe(recipe));
+        if (ingredient.getItem() == Items.potionitem || ingredients.contains(ingredient))
+            for (NEIBrewingRecipe recipe : apotions)
+                if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.input.item, ingredient))
+                    arecipes.add(new CachedBrewingRecipe(recipe));
+
+        label0: for (IBrewingRecipe recipe : BrewingRecipeRegistry.getRecipes())
+            if (recipe instanceof BrewingRecipe || recipe instanceof BrewingOreRecipe)
+                if (NEIServerUtils.areStacksSameType(((AbstractBrewingRecipe<?>) recipe).input, ingredient))
```
Comment (Chicken-Bones) [inline src/codechicken/nei/recipe/BrewingRecipeHandler.java]: The AbstractBrewingRecipe<?> cast is done 4 times, it's a bit verbose, make a local variable, perhaps named arecipe

**Comment by Chicken-Bones** [comment]
Just a few minor things. Thanks for this and good job on following my code style.

**Comment by ganymedes01** [comment]
No problem!
Done the changes you asked :3

File: src/codechicken/nei/recipe/BrewingRecipeHandler.java [hunk 2]
```diff
@@ -83,30 +97,52 @@ public String getRecipeName() {
...
-
-        for (BrewingRecipe recipe : apotions)
-            if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.precursorPotion.item, ingredient))
-                arecipes.add(new CachedBrewingRecipe(recipe));
+        if (ingredient.getItem() == Items.potionitem || ingredients.contains(ingredient))
+            for (NEIBrewingRecipe recipe : apotions)
+                if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.input.item, ingredient))
+                    arecipes.add(new CachedBrewingRecipe(recipe));
+
+        label0: for (IBrewingRecipe recipe : BrewingRecipeRegistry.getRecipes())
```
Comment (Chicken-Bones) [inline src/codechicken/nei/recipe/BrewingRecipeHandler.java]: Missed the label0 change? nextRecipe perhaps

**Comment by ganymedes01** [comment]
Oh sorry, forgot to do that one.
Anything else ? :3

**Pull request merged**